src/VisualStudio/Core/Def/Implementation/PdbSourceDocument/PdbSourceDocumentOutputWindowLogger.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PdbSourceDocument;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.PdbSourceDocument
+{
+    [Export(typeof(IPdbSourceDocumentLogger)), Shared]
+    internal sealed class PdbSourceDocumentOutputWindowLogger : IPdbSourceDocumentLogger
+    {
+        private static readonly Guid s_outputPaneGuid = new Guid("f543e896-2e9c-48b8-8fac-d1d5030b4b89");
+        private IVsOutputWindowPane? _outputPane;
+
+        private readonly IThreadingContext _threadingContext;
+        private readonly IServiceProvider _serviceProvider;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public PdbSourceDocumentOutputWindowLogger(SVsServiceProvider serviceProvider, IThreadingContext threadingContext)
+        {
+            _serviceProvider = serviceProvider;
+            _threadingContext = threadingContext;
+        }
+
+        public void Clear()
+        {
+            _threadingContext.JoinableTaskFactory.Run(async () =>
```

Comment:
Should these methods be async so we're not potentially blocking up a background thread?

Reply:
I also worry about deadlock risk here if we end up with a UI thread blocking on the call to GetGeneratedFileAsync, and then this trying to hop back. Given this is only logging, should this just be queuing work to the UI thread and we keep the actual updating of the pane asynchronous? It'd remove the deadlock risk, and also would probably help performance so you're not doing so many UI transitions during the main code path.